src/Features/CSharp/Portable/CodeFixes/Async/CSharpAddAwaitCodeFixProvider.cs

```diff
@@ -76,8 +72,19 @@ private bool IsCorrectReturnType(ExpressionSyntax expression, SemanticModel sema
 
         private static ExpressionSyntax ConvertToAwaitExpression(ExpressionSyntax expression)
         {
-            return SyntaxFactory.AwaitExpression(expression)
-                                .WithAdditionalAnnotations(Formatter.Annotation);
+            AwaitExpressionSyntax result;
+
+            if (expression.HasLeadingTrivia)
```

Comment:
There is no case where we don't want to propagate trivia.  What if there is a comment after this expression?

```C#
return SyntaxFactory.AwaitExpression(expression)
        .WithTriviaFrom(expression)
        .WithAdditionalAnnotations(Formatter.Annotation);
```